cmake/P4CUtils.cmake

```diff
@@ -119,18 +119,21 @@ endmacro(p4c_add_test_label)
 # The macro generates the test files in a directory prefixed by tag.
 #
 macro(p4c_add_test_list tag driver tests xfail)
-  set (__xfail_list "${xfail}")
-  set (__test_list "${tests}")
+  # Sanitize the input lists to account for mixed path input
+  p4c_set_abs_path("${xfail}" __xfail_list)
```

Comment:
@cc10512 : can you please take a look at the changes in this file?
I don't know of all the uses of the build process to correctly review this file.

Reply:
I am also happy if we find a better way to make sure that all paths have the same root folder. Right now, I am casting to an absolute path and then back to a relative one. That does the job, but it is not great.
Oh and in `p4c_add_tests` only the "testsuites" list is checked with `p4c_find_test_names`, "xfail" is passed along as is. I was sure if that is intentional.

Reply:
As you correctly state, each element in the "tests" list is relative to P4C_SOURCE_DIR. The xfails are based on whatever you write as xfail in your CMakeLists. We typically write them based on P4C_SOURCE_DIR as well. For example: 'testdata/p4_16_samples/issue774-4-bmv2.p4' or 'extensions/<extension_name>/p4_16/ingress_checksum.p4'. 
I suppose you could normalize them all to an absolute path. However, since you're converting back to relative paths, why not simply convert the xfails to relative paths based on P4C_SOURCE_DIR and leave the tests as specified?

Reply:
I think I remember the problem I was dealing with. If the input is already a relative path, which is often the case, the cmake function `file RELATIVE PATH` throws an error:
`file RELATIVE_PATH must be passed a full path to the file`.
Which is why I first converted all inputs to a full path. If I modify `p4c_find_test_names` to use the absolute path instead of a relative one, no test gets added at all. Likely because of some specific behaviour in `p4c_add_test`?

Reply:
Likely. We do name the tests based on the path unless you provide an alias, and in that case cmake may barf. I don't recall, but if you attach a log, I could take a look. However, I do think that if you just convert xfails to absolute paths and then back to relative w.r.t. P4C_SOURCE_DIR, you should be in good shape. That is, you don't need the absolute path comparison, you just seem to need to converge on one convention. The one we've been following is to provide the correct relative path for elements in the xfail list. Moreover, if you have multiple tests with the same name in different directories, you'll have to provide enough of their path in the xfail spec to distinguish. So, overall, I think you're better off giving the full relative path!